Objects/object.c

```diff
@@ -887,10 +887,42 @@ PyObject_GetAttr(PyObject *v, PyObject *name)
     return NULL;
 }
 
+PyObject *
+_PyObject_GetAttrWithoutError(PyObject *v, PyObject *name)
+{
+    PyTypeObject *tp = Py_TYPE(v);
+    PyObject *ret = NULL;
+
+    if (!PyUnicode_Check(name)) {
+        PyErr_Format(PyExc_TypeError,
+                     "attribute name must be string, not '%.200s'",
+                     name->ob_type->tp_name);
+        return NULL;
+    }
+    if (tp->tp_getattro != NULL) {
+        if (tp->tp_getattro == PyObject_GenericGetAttr) {
+            ret = _PyObject_GenericGetAttrWithDict(v, name, NULL, 1);
```

Comment:
Can `_PyObject_GenericGetAttrWithDict()` raise an AttributeError? If can't, it is better to just return the result here. `PyErr_ExceptionMatches(PyExc_AttributeError)` has a non-zero cost, in some cases it can be significant. For example see bpo-31336.

Reply:
Yes, it may call descriptor.  Any exception can be raised.
`suppress=1` only suppress AttributeError raised from _PyObject_GenericGetAttrWithDict
directly.

Reply:
I changed _PyObject_GenericGetAttrWithDict() to suppress AttributeError
from descriptors too.
Since _PyObject_GenericGetAttrWithDict is performance critical function,
I'll run benchmark suite before merge this.